cpp/build-support/lint_cpp_cli.py

```diff
@@ -77,6 +77,9 @@ def lint_file(path):
 
 EXCLUSIONS = _paths('''\
     arrow/arrow-config.cmake
+    arrow/flight/sql/odbc/flight_sql/get_info_cache.h
+    arrow/flight/sql/odbc/odbcabstraction/include/odbcabstraction/blocking_queue.h
+    arrow/flight/sql/odbc/odbcabstraction/include/odbcabstraction/odbc_impl/odbc_handle.h
```

Comment:
Why do we need them?
Are they copied from somewhere?

Reply:
Ah, OK. Let's work on #45810 (remove CLI/C++ support) before this.

Do you want to work on #45810 or do you want me to work on #45810?